x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/analytics/event/AnalyticsEventParser.java

```diff
@@ -0,0 +1,299 @@
+/*
```

Comment:
I would probably have a class per event type, and create a factory on top of it that parses payload and creates an object accordingly.